server/war/src/main/webapp/WEB-INF/web.xml

```diff
@@ -33,6 +33,8 @@
     <web-resource-collection>
       <web-resource-name>hawkular-apm-server</web-resource-name>
       <url-pattern>/*</url-pattern>
+      <!-- allow OPTIONS for CORS -->
+      <http-method-omission>OPTIONS</http-method-omission>

```

Comment:
IIRC this is saying that OPTIONS method is not secured. This allows CORS preflight request to pass to the CorsFilter. Otherwise security constraints will reject the request.

@jpkrohling could you please look at this. I was not able to make it work without this, however maybe there is a way haw to add security data to preflight request.

This is the example code which needs CORS filter. hawkular/hawkular-apm-opentracing-javascript#11